src/main/java/wayoftime/bloodmagic/common/block/BlockSpectral.java

```diff
@@ -30,22 +32,14 @@ public BlockSpectral(Properties prop)
 		super(prop);
 	}
 
-	public void tick(BlockState state, ServerLevel world, BlockPos pos, Random rand)
-	{
-		switch (state.getValue(SPECTRAL_STATE))
-		{
-		case SOLID:
-			world.setBlock(pos, state.setValue(SPECTRAL_STATE, SpectralBlockType.LEAKING), 3);
-			world.scheduleTick(pos, this, BlockSpectral.DECAY_RATE);
-			break;
-		case LEAKING:
-			BlockEntity tile = world.getBlockEntity(pos);
+	@Override
+	public <T extends BlockEntity> BlockEntityTicker<T> getTicker(Level level, BlockState state, BlockEntityType<T> type) {
+		return (level1, blockPos, blockState, tile) -> {
 			if (tile instanceof TileSpectral)
 			{
-				((TileSpectral) tile).revertToFluid();
+				((TileSpectral) tile).tick();
```

Comment:
you should pass the blockPos, level and tile from the lambda into the `tick` method instead of fetching them in the method. Also use the automatic cast when using `instanceof` instead of manual casting